inc/Remote/JsonRpcServer.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace dokuwiki\Remote;
+
+/**
+ * Provide the Remote XMLRPC API as a JSON based API
+ */
+class JsonRpcServer
+{
+
+    protected $remote;
+
+    /**
+     * JsonRpcServer constructor.
+     */
+    public function __construct()
+    {
+        $this->remote = new Api();
+        $this->remote->setFileTransformation(array($this, 'toFile'));
+    }
+
+    /**
+     * Serve the request
+     *
+     * @return mixed
+     * @throws RemoteException
+     */
+    public function serve()
+    {
+        global $conf;
+        if (!$conf['remote']) {
+            http_status(404);
+            throw new RemoteException("JSON-RPC server not enabled.", -32605);
+        }
+        if (!empty($conf['remotecors'])) {
+            header('Access-Control-Allow-Origin: ' . $conf['remotecors']);
+        }
+
+        global $INPUT;
+
+        $call = $INPUT->server->str('PATH_INFO');
+        $call = trim($call, '/');
+        $args = json_decode(file_get_contents('php://input'), true);
+        if (!is_array($args)) $args = [];
+
+        return $this->call($call, $args);
+    }
+
+    /**
+     * Call an API method
+     *
+     * @param string $methodname
+     * @param array $args
+     * @return mixed
+     * @throws RemoteException
+     */
+    public function call($methodname, $args)
+    {
+        try {
+            $result = $this->remote->call($methodname, $args);
+            return $result;
+        } catch (AccessDeniedException $e) {
+            if (!isset($_SERVER['REMOTE_USER'])) {
+                http_status(401);
+                throw new RemoteException("server error. not authorized to call method $methodname", -32603);
```

Comment:
nice, i was confused why xmlrpc did return rather throw, as it resulted the exceptions never to be seen by the client.

but I never got around reporting/questioning it, as I changed xmlrpc library and using throw didn't make them to be seen any way.

morale of this post: are those errors seen as client responses? perhaps add example to pr body.